sycl/test/basic_tests/range_ctad.cpp

```diff
@@ -0,0 +1,29 @@
+// RUN: %clangxx -std=c++17 -fsyntax-only -Xclang -verify %s
+// expected-no-diagnostics
+//==--------------- range_ctad.cpp - SYCL range CTAD test ----------------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+#include <CL/sycl.hpp>
+
+using namespace std;
+int main() {
+  cl::sycl::range one_dim_range(64);
+  cl::sycl::range two_dim_range(64, 1);
+  cl::sycl::range three_dim_range(64, 1, 2);
```

Comment:
Interesting to combine new-style CTAD with old-style explicit constructor call...